auth/role.go

```diff
@@ -139,6 +150,14 @@ func (role *roleImpl) SetCas(cas uint64) {
 	role.cas = cas
 }
 
+func (role *roleImpl) setDeleted(deleted bool) {
+	role.Deleted = deleted
+}
+
+func (role *roleImpl) IsDeleted() bool {
```

Comment:
It was unexpected to me that the read accessor is public, but the write accessor is private. Is this intentional?

Reply:
Similarly, needs to be read in the https://github.com/couchbase/sync_gateway/pull/4986/files#diff-c2dab3b01f62851ce6742b56d378463c64d11ed2a1f1d9da5055781e819c3a8cR231 case